### DISCUSSION TITLE
Total votes reward

### DISCUSSION THREAD

**Discussion by daoshii** [Zrips/CMI#8096] [Enhancement]
Voting rewards are already built in per player.
But total votes every time the server receives X amount votes, Their would be a vote party which runs commands.

Ex. "Server has received another 50 votes! Here's 5 more vote keys for everyone and a party pinata to smash!"

**Comment by CloudeLecaw** [comment]
Hmm so cmi votes should become the new Voteparty? XD

By the way we allready made this what you wanted on our server with CMI vote commands (config) and useage of CMI Specialzed Commands :)
We defined some placeholder that counts up with each vote and these votes can react on our defined conditions.
Here you see a lite paste of our Vote section from config.y,ö

```
  CommandsOnVote:
  - cmi kit Votis [playerName] -s
  - cmi ifonline [playerName] cmi actionbarmsg [playerName] -s:5 &6Einen Voti erhalten für das Voten auf &e[serviceName]&6.
  - cmi ifonline [playerName] cmi ctellraw [playerName] <T>!&6Für das Voten auf &e[serviceName] &6hast du &eeinen Voti &6erhalten.</T>
  - cmi ifonline [playerName] aach add 1 Custom.votes [playerName]
  - phc setsilent totalservervotes %cmi_equationint_{phc_totalservervotes}+1%
  - phc setsilent totalservervotes30day %cmi_equationint_{phc_totalservervotes30day}+1%
  - phc setsilent totalservervotes30000 %cmi_equationint_{phc_totalservervotes30000}+1%
  - check:%phc_totalservervotes30000%==10000! cmi money all 1000 -s
  - check:%phc_totalservervotes30000%==10000! cmi money alloffline 1000 -s
  - check:%phc_totalservervotes30000%==20000! cmi money all 1000 -s
  - check:%phc_totalservervotes30000%==20000! cmi money alloffline 1000 -s
  - check:%phc_totalservervotes30000%==30000! cmi money all 1000 -s
  - check:%phc_totalservervotes30000%==30000! cmi money alloffline 1000 -s
  - check:%phc_totalservervotes30000%==30001! phc setsilent totalservervotes30000 1
 ```

- The Votes get counted by the placeholder for allvotes,30days and 30000 votes.
- Wehn the 30000 count placeholder reaches the value 30001 it get set to 1 (a reset)
- The 30day placeholder counts all monthly Votes (This placeholder got reseted with CMIs Schedule on each 1st of month a secound after midnight.)

PHC is our own private plugin to create and manage own placeholders with own values that can be used everywhere where you have PlaceholderAPI support :-)

You see at least its possible allready with the help of a little 3rd plugin.
Just CMI alone cant do this right now in a standalone because you cant create own public placeholders you could messaure with CMI.

**Reply by daoshii** [reply]
CMI doesn't need custom placeholders for a voteparty, and 90% of the logic is already built in. Players are currently awarded individually for reaching X amount of votes in config.

The entirety of this suggestion is to eliminate another plugin, as CMI has done many times in the past, rather than add another one, whatever PHC is.

**Reply by CloudeLecaw** [reply]
All CMI would need is a placeholder that counts all votes ever together

The rest of your voteparty like plugin can be done allready like you noticed if you looked at my commandrewards i showed you above...   cmi can just not do 90% allready thanks to specialzed commands it can do 96%, because just one placeholder is missing :D   Soo may you should request the CMI Placeholder?  The Informations are indirectly stored allready in database
It just needs a placeholder like may %cmi_votestotal" that count all playervotes from the database together.

**Reply by daoshii** [reply]
No it does not need a placeholder, what are you even talking about? A placeholder would be an extension for logic that exists within the plugin. Im not asking for a placeholder, go somewhere else.

**Comment by CloudeLecaw** [comment]
Alaso may see allready created discussions about Vote extansions:

https://github.com/Zrips/CMI/discussions/3021 <-  request: rewards to the player for all X total votes
https://github.com/Zrips/CMI/discussions/7876 <-  request: Commands Running Every X Votes
https://github.com/Zrips/CMI/discussions/6534 <-  request: keep votetrack of other voteplugins plugins
https://github.com/Zrips/CMI/discussions/3118 <-  request: voteparty implementation
https://github.com/Zrips/CMI/discussions/3945 <-  request: extra votecount and voterewards
https://github.com/Zrips/CMI/discussions/6997 <-  request: hold voterewards until player logs in
https://github.com/Zrips/CMI/discussions/5856 <-  request: random reward from a list, vote streaks
https://github.com/Zrips/CMI/discussions/4335 <-  request: voteparty features implementation
https://github.com/Zrips/CMI/discussions/6864 <-  request: randomize the voting rewards
https://github.com/Zrips/CMI/discussions/5204 <-  request: voteparty implementation
https://github.com/Zrips/CMI/discussions/4681 <-  request for: votes:[amount]! and todayvotes:[amount]! for checks
https://github.com/Zrips/CMI/discussions/2239 <-  request: Voting reminder
https://github.com/Zrips/CMI/discussions/1203 <-  request: randomize the voting rewards
https://github.com/Zrips/CMI/discussions/3357 <-  request: rewards to the player for all X votes
https://github.com/Zrips/CMI/discussions/3857 <-  request: daylie, weekly, monthly votecounters
https://github.com/Zrips/CMI/discussions/3202 <-  request: voteparty implementation

if your looking at them you might also find your requested things more then 2-3 times...

Ohh by the way.. an it doesnt matter if a request discussion is 4 years or only week old.
Everyone can still add some new lines to it and upvote it so it might get seen sooner..
But nooo.. people are to lazzy to search, if its easiert to make new once..

**Reply by daoshii** [reply]
One from 2020 with no reply. Another from 2020 with one reply from Floris. 6534 has absolutely nothing to do with a voting party. So congrats on finding 1 similar, recent thread. What are you the CMI police? Lay off, and your advice is absolute dog water, along with your plugins that you constantly suggest others use on CMI discussion threads that have nothing to do with your dog water plugins.

**Reply by CloudeLecaw** [reply]
```
                          ________________
                          \      __      /         __
                           \_____()_____/         /  )
                           '============`        /  /
                            #---\  /---#        /  /
                           (# @\| |/@  #)      /  /
                            \   (_)   /       /  /
                            |\ '---` /|      /  /
                    _______/ \\_____// \____/ o_|
                   /       \  /     \  /   / o_|
                  / |           o|        / o_| \
                 /  |  _____     |       / /   \ \
                /   |  |===|    o|      / /\    \ \
               |    |   \@/      |     / /  \    \ \
               |    |___________o|__/----)   \    \/
               |    '              ||  --)    \     |
               |___________________||  --)     \    /
                    |           o|   ''''   |   \__/
                    |            |          |
```

**Reply by daoshii** [reply]
Why are you still replying when im obviously ignoring you. Seriously find something to do other than necro peoples discussions.

And wow.. you edited your post and added more threads.. What are you still doing here? This is not your post seriously find something else to do..

**Comment by CloudeLecaw** [comment]
it was not discussed with anymore - might means only one of the 1000s CMI users is interested in it-> results in low prioirty or in beeing forgotten.
you think here it will be better, if all people allways just make new discussion instead of discuss in an might be older one?
If there are so many requests and no reply or other dissussion about it was given it might has a reason? 😅
one because people only make new requests instead of showing zrips that more people are interested in a discussion in one discussion.

Example: 20 people from same city want to discuss about an may starting event in their city.
Nr1 makes chatroom 1 and start writing and waits for responce, Nr2 makes chatroom 2 and start writing and waits for responce, Nr3 makes chatroom 3 and start writing and waits for responce,  Nr4 makes chatroom 4 and start writing and waits for responce..........Nr20 makes chatroom 20 start writing and waits for responce,    What you think will be the result of this at the end? ;)
The same is here on CMI Github. many people may want things but instead of search for existing discussion and add some lines that they are also instered in it they are mostly make new discussions with the result of noone will answer.

**Reply by daoshii** [reply]
Who's "us"? I wasn't showing you anything. You just seem to think I care what you have to say, regardless of how unintelligent your 'work arounds' are, even though I never asked for a work around.

Every CMI discussion - "Hey can I get some parmesan for my spaghetti?"
CloudeLecaw - "Why don't you just try my plugins instead"

**Reply by TheBadshah** [reply]
Guess who don't know meaning of discussion

**Reply by CloudeLecaw** [reply]
At least i only said just a placeholder is missing for cmi for total count of all votes to relaise the thing that you requested.
The rest you can done with cmis specilzed commands allready.
Because of the missing of this placeholder and may the reason votesection allways can benefit from new extansions, it could be that this also is a reason why i upvoted your discussion some days before allready.
may you realize that? :D
![image](https://github.com/Zrips/CMI/assets/12080017/9ede337a-a331-4edc-b1c3-60e14ea405cd)
As more people discuss and upvote a discussion as higher is the chance that something (maybe not excatlcy the requests thing) will be in the distant future update ;-)


> Every CMI discussion - "Hey can I get some parmesan for my spaghetti?"
CloudeLecaw - "Why don't you just try my plugins instead"

hmm i dont share my private plugin xD but instead of spaghetti how about tortellinis? They have a much better taste :D

hmm daoshi can turn on two candels at the end of the day ;) one for TheBadshah because he had the time for a quick responce and one for the little man that asked for tortellinis instead of spaghetti <3

**Reply by daoshii** [reply]
Wow badshah has participated in 1 discussion on CMI, he speaks exactly like you with non-fluent english... i wonderrr..........

**Comment by CloudeLecaw** [comment]


**Reply by mrfloris** [reply]
Can you refrain from posting ascii things like those. It's not constructive and off topic, and gives the wrong impression. 

It looks like they've requested to discontinue the conversation with you. Let's respect that. If they don't want to get your information or don't understand it. That's on them, not you to try and convince them.

**Reply by CloudeLecaw** [reply]
well i removed this one :D

Year looks like he requested this.. but at least he still replies ;)

**Comment by Momshroom** [comment]
Please stop being rude to the CMI staff.   It’s not going to get you help
and they are trying to help you.  (Yes, I know I’m just a rando, but it’s
painful to watch volunteers getting bashed.)

**Reply by daoshii** [reply]
Are you not reading the conversation? I simply said im not interested in what he has to offer. So why continue to attack me about it, and also get on alternate accounts or drag friends in to try to support his cause?

If I say im not interested then im not interested... People need to get that hint. He was the one who started being rude and im just expected to sit back and take it solely because hes "staff"? Not a very good image.

**Reply by daoshii** [reply]
On top of that, he has made edit to every single comment, 10+ times on some of them to make himself look good. Must not be aware we can see edit history.

**Comment by mrfloris** [comment]
@op and cloud, etc. Let's get back on topic. The Discussion was published with a request to enhance the features. Information has been shared and more details kinda discussed. Let's avoid getting personal or dramatic. We can leave this chat as is. Thank you.

**Comment by CloudeLecaw** [comment]
You can make a Reward every X Votes of a player allready with CMI

in the CMI Votecommand use a bit of usermeta and specialzed command: 
Add this to your Voterewadscommands in config:

```
- cmi usermeta [playerName] increment voterewardevery30votes 1 -s
- delay! 1
- check:%cmi_user_metaint_voterewardevery30votes%==30! cmi money give [playername] 1000 -s
- check:%cmi_user_metaint_voterewardevery30votes%==30! cmi usermeta [playerName] remove voterewardevery30votes
```
This gives extra 1000 money each 30 votes a player has made and rewards him with1000 economy dollars
after that the meta got reset. you can change the needed amount by change the number behind the ==
you can check the amount with the placeholder %cmi_user_metaint_voterewardevery30votes% or with the command
cmi usermeta PLAYERSNAME list



what cant be done with CMI currently is the Voteparty (overall votes)
Therefore a placeholder would be need that count all votes and also a way to set/reset with cmi commands?

But what can be sais is that at least the reward every x votes a player did cant be done allready :-)
(to be honest im not sure if this check about the placeholder of cmi placeholders also works if the player if offline...)
